lib/affirm/charge_event.rb

```diff
@@ -0,0 +1,20 @@
+module Affirm
+  class ChargeEvent
+    attr_reader :id, :transaction_id, :type, :created,
+      :amount, :fee, :order_id, :shipping_carrier, :shipping_confirmation,
+      :fee_refunded
+
+    def initialize(attrs)
+      @id                    = attrs["id"]
+      @transaction_id        = attrs["transaction_id"]
+      @type                  = attrs["type"]
+      @created               = attrs["created"]
+      @amount                = attrs["amount"]
+      @fee                   = attrs["fee"]
```

Comment:
Ah I see. 

I mean, it's not a pattern I see a lot, which doesn't say that it's better or worse. I guess the principle would be to create methods so that it's only doing the action when it's needed. Otherwise, we're creating ivars or parsing input that never needs to get parsed because that method is never called.